traffic_ops/ort/atstccfg/toreq/caching.go

```diff
@@ -105,24 +125,42 @@ func GetJSONObjFromFile(tempDir string, cacheFileName string, cacheFileMaxAge ti
 	if err != nil {
 		return errors.New("getting object file info '" + objPath + "':" + err.Error())
 	}
-
 	if objFileAge := time.Now().Sub(objFileInfo.ModTime()); objFileAge > cacheFileMaxAge {
 		return fmt.Errorf("object file too old, max age %dms less than file age %dms", int(cacheFileMaxAge/time.Millisecond), int(objFileAge/time.Millisecond))
 	}
-
-	bts, err := ioutil.ReadAll(objFile)
-	if err != nil {
-		return errors.New("reading object from file '" + objPath + "':" + err.Error())
-	}
-
-	if err := json.Unmarshal(bts, obj); err != nil {
+	if err := decode(objFile, obj); err != nil {
```

Comment:
Likewise with the decoder. Should a nil encoder have a default meaning?

Reply:
Changed to return an error, as above